insights/core/plugins.py

```diff
@@ -413,6 +414,8 @@ def register_component(
         EMITTERS.add(component)
         register_consumer(component)
 
+    fava.add_shared_parser(component.__name__, component)
```

Comment:
Not sure about this, but you @csams would know better than me.  Do all components end up with key/value pairs in the 'shared' argument? Or only some components?   Fava can only handle components that show up in the 'shared' argument.   Maybe this line needs an 'if component.shared:' protecting it?

Reply:
@gavin-romig-koch You're right, we should protect that. The registration function has a boolean for whether the component is shared, so I'll use that to guard.